### PR TITLE
Buzzer - Added Support for Buzzer for BBBMini and lost vehicle event handler

### PR DESCRIPTION
**Pull request by juvinski**
Added support for BBB Black/Green GPIO P8_12(GPIO44) and Lost Vehicle support

### CODE REVIEW

File: libraries/AP_Notify/AP_Notify.cpp [hunk 1]
```diff
@@ -99,8 +99,9 @@ struct AP_Notify::notify_events_type AP_Notify::events;
         ToshibaLED_I2C toshibaled;
         NotifyDevice *AP_Notify::_devices[] = {&navioled, &toshibaled};
     #elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BBBMINI
+        Buzzer buzzer; //Added support for buzzer
```
Comment (OXINARF) [inline libraries/AP_Notify/AP_Notify.cpp]: No need for that comment in code

File: libraries/AP_Notify/Buzzer.cpp [hunk 2]
```diff
@@ -32,6 +33,7 @@ bool Buzzer::init()
 
     // setup the pin and ensure it's off
     hal.gpio->pinMode(BUZZER_PIN, HAL_GPIO_OUTPUT);
+
```
Comment (OXINARF) [inline libraries/AP_Notify/Buzzer.cpp]: Cosmetic changes should be done in a separate commit, but these two don't seem necessary, leftovers?

File: libraries/AP_Notify/Buzzer.cpp [hunk 3]
```diff
@@ -184,7 +186,12 @@ void Buzzer::update()
         }
         return;
     }
-
+
```
Comment (OXINARF) [inline libraries/AP_Notify/Buzzer.cpp]: Not so important, but empty lines should be really empty (no spaces or tabs).

File: libraries/AP_Notify/Buzzer.h [hunk 4]
```diff
@@ -18,11 +18,16 @@
...
  # define BUZZER_PIN    32
+#elif CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+ #if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BBBMINI
+    #define BUZZER_PIN     44
+ #else
+    #define BUZZER_PIN     0
+ #endif
 #else
- # define BUZZER_PIN    0       // pin undefined on other boards
+    #define BUZZER_PIN     0
```
Comment (OXINARF) [inline libraries/AP_Notify/Buzzer.h]: Why remove the comment? There's no need to duplicate it to the other zero, just in this one is fine.

File: libraries/AP_Notify/Buzzer.h [hunk 5]
```diff
@@ -66,11 +72,12 @@ class Buzzer: public NotifyDevice
 
     /// buzzer_flag_type - bitmask of current state and ap_notify states we track
     struct buzzer_flag_type {
+        uint8_t vehicle_lost        : 1;     //1 Vehicle Lost
```
Comment (OXINARF) [inline libraries/AP_Notify/Buzzer.h]: I don't think this is actually needed, but it does follow the way it was implemented (there's already other vars that weren't needed)

**Comment by OXINARF** [comment]
Besides the small comments I've made, I think that you should squash your commits.

**Comment by juvinski** [comment]
hi @OXINARF ,

thanks a lot for the comments, I will review and remove the necessary things , remove the comments and squash previous in one final commit.
Thanks

**Comment by mirkix** [comment]
I will check the PR tomorrow.

**Comment by mirkix** [comment]
@juvinski Thanks for this PR but P8_12 is already in use https://github.com/mirkix/BBBMINI-PCB/blob/master/schematic/bbbmini.pdf . I will propose a new pin for the buzzer that can be used for the BBBmini.

**Comment by juvinski** [comment]
Hi @mirkix , I forgot to look at schematic. any suggestion of what woulb be a better pin?
Regards

**Comment by mirkix** [comment]
@juvinski Can you try and use P8_31.

**Comment by juvinski** [comment]
Hi @mirkix, I changed and tested pin 8.31 and it's working.
@OXINARF, I updated code.

**Comment by OXINARF** [comment]
@juvinski You shouldn't have merge commits (you would need to rebase instead of merging) and all these commits can be squashed into one or two. Also, the commit messages now mention the wrong pin. Also you still left some unneeded cosmetic changes.
Since this is your first PR, I think all of this can be fixed by @lucasdemarchi when bringing it in.

@mirkix Can you it a final review? Thanks.

**Comment by lucasdemarchi** [comment]
Fixed the comments and sent a new version: #4095
